File: src/main/java/de/tectoast/emolga/utils/sql/managers/GiveawayManager.java
package de.tectoast.emolga.utils.sql.managers;

import de.tectoast.emolga.utils.Giveaway;
import de.tectoast.emolga.utils.sql.base.DataManager;
import de.tectoast.emolga.utils.sql.base.columns.IntColumn;
import de.tectoast.emolga.utils.sql.base.columns.LongColumn;
import de.tectoast.emolga.utils.sql.base.columns.StringColumn;
import de.tectoast.emolga.utils.sql.base.columns.TimestampColumn;

public class GiveawayManager extends DataManager {

    final LongColumn MESSAGEID = new LongColumn("messageid", this);
    final LongColumn CHANNELID = new LongColumn("channelid", this);
    final LongColumn HOSTID = new LongColumn("hostid", this);
    final StringColumn PRIZE = new StringColumn("prize", this);
    final TimestampColumn END = new TimestampColumn("end", this);
    final IntColumn WINNERS = new IntColumn("winners", this);

    public GiveawayManager() {
        super("giveaways");
        setColumns(MESSAGEID, CHANNELID, HOSTID, PRIZE, END, WINNERS);
    }

    public void saveGiveaway(Giveaway g) {
        insert(g.getMessageId(), g.getChannelId(), g.getUserId(), g.getPrize(), g.getEnd(), g.getWinners());
    }

    public void removeGiveaway(Giveaway g) {
        delete(MESSAGEID.check(g.getMessageId()));
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/sql/managers/GiveawayManager.kt
package de.tectoast.emolga.utils.sql.managers

import de.tectoast.emolga.utils.Giveaway
import de.tectoast.emolga.utils.sql.base.DataManager
import de.tectoast.emolga.utils.sql.base.columns.IntColumn
import de.tectoast.emolga.utils.sql.base.columns.LongColumn
import de.tectoast.emolga.utils.sql.base.columns.StringColumn
import de.tectoast.emolga.utils.sql.base.columns.TimestampColumn

object GiveawayManager : DataManager("giveaways") {
    private val MESSAGEID = LongColumn("messageid", this)
    private val CHANNELID = LongColumn("channelid", this)
    private val HOSTID = LongColumn("hostid", this)
    private val PRIZE = StringColumn("prize", this)
    private val END = TimestampColumn("end", this)
    private val WINNERS = IntColumn("winners", this)

    init {
        setColumns(MESSAGEID, CHANNELID, HOSTID, PRIZE, END, WINNERS)
    }

    fun saveGiveaway(g: Giveaway) {
        insert(g.messageId, g.channelId, g.userId, g.prize, g.end, g.winners)
    }

    fun removeGiveaway(g: Giveaway) {
        delete(MESSAGEID.check(g.messageId))
    }
}